Fix infinite recursion in API BC test

aorenste opened the pull request:
```
python test/test_fx.py -k test_public_api_surface
```
was failing with a complaint about infinite recursion. Fixed that and then marked the two API changes from #123681 as private (for `get_example_value`) and backward compatible (for `insert_deferred_runtime_asserts`).

Fixes #104012 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125706

aorenste committed:
Fix infinite recursion in API BC test

[ghstack-poisoned]

aorenste commented:
@BoyuanFeng Can you verify that these two entrypoints should be considered backward compatible?

ezyang commented on torch/fx/passes/runtime_assert.py:
```diff
@@ -14,11 +14,13 @@
 
 
+@compatibility(is_backward_compatible=True)
 def get_example_value(node: fx.Node) -> Optional[str]:
```
We can land this as is to unblock but this function should be private

aorenste replied:
I need to update anyway since, as usual, I forgot to run lintrunner before submitting...

aorenste committed:
Update on "Fix infinite recursion in API BC test"

```
python test/test_fx.py -k test_public_api_surface
```
was failing with a complaint about infinite recursion. Fixed that and then marked the two API changes from #123681 as backward compatible.

Fixes #104012 




[ghstack-poisoned]

aorenste commented:
@pytorchbot merge

aorenste committed:
Update on "Fix infinite recursion in API BC test"

```
python test/test_fx.py -k test_public_api_surface
```
was failing with a complaint about infinite recursion. Fixed that and then marked the two API changes from #123681 as private (for `get_example_value`) and backward compatible (for `insert_deferred_runtime_asserts`).

Fixes #104012 




[ghstack-poisoned]

aorenste commented:
@pytorchbot merge
